packages/components/src/Modal/CloseHeaderButton.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2021 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import { useTranslation } from 'react-i18next'
+import React, { FC, useContext } from 'react'
+import { Close } from '@styled-icons/material/Close'
+import styled from 'styled-components'
+import { IconButton, IconButtonProps } from '../Button'
+import { DialogContext } from '../Dialog'
+
+export interface CloseHeaderButtonProps {
+  /**
+   * this will define the size of the button
+   * @default 'medium'
+   */
+  size?: IconButtonProps['size']
+}
+
+const CloseHeaderButtonLayout: FC<CloseHeaderButtonProps> = ({
+  size = 'medium',
+}) => {
+  const { t } = useTranslation('CloseHeaderButton')
+  const { closeModal, id } = useContext(DialogContext)
+  return (
+    <IconButton
+      id={id ? `${id}-iconButton` : undefined}
+      size={size}
+      onClick={closeModal}
+      label={t('Close')}
+      icon={<Close />}
+    />
+  )
+}
+
+export const CloseHeaderButton = styled(
+  CloseHeaderButtonLayout
+)<CloseHeaderButtonProps>``
```

Comment:
See name suggestion above again. Also, I don't think the `CloseHeaderButtonProps` needs to be specified because Styled Components _should_ infer it automatically.

```suggestion
export const ModalHeaderCloseButton = 
  styled(ModalHeaderCloseButtonLayout)``
```